### ISSUE TITLE
[BUG] python function flavor load_model not working from all directories on local file system

### ISSUE THREAD

**Issue by vibss2397** [mlflow/mlflow#4615]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).

**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [- ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS
- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: 1.18.0
- **Python version**: 3.7.10
- **npm version, if running the dev UI**: None
- **Exact command to reproduce**:

### Describe the problem
So I have created a basic model in the ml pyfunc flavor, which looks something like :
```
class MyModel(mlflow.pyfunc.PythonModel):
    def __init__(self, n_components=20, random_state=42):
        self.n_components = n_components
        self.random_state = random_state
    
    def predict(self, context, model_input):
        n_samples = model_input.shape[0]
        return pd.DataFrame(np.random.randn(n_samples)) + self.n_components
```
I then log this model to a tracking server. Now when i try to load this model using 👍 
```
mlflow.pyfunc.load_model(model_uri)
```
the model loads in this directory. But when i move out of this directory and run the same command, I get an error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/anaconda3/envs/butter/lib/python3.7/site-packages/mlflow/pyfunc/__init__.py", line 655, in load_model
    model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
  File "/opt/anaconda3/envs/butter/lib/python3.7/site-packages/mlflow/pyfunc/model.py", line 226, in _load_pyfunc
    python_model = cloudpickle.load(f)
ModuleNotFoundError: No module named 'models'
```
However, If i again log a model from the outside directory and i load again, it starts working. I am not sure if i need to set a directory to cache models somewhere, if i run it from a new directory.

Tl;dr : The same command to load a pyfunc model works in a directory but not in other directories. What are the possible fixes to this.

### Code to reproduce issue
1. log a model to an s3 bucket/locally and then try to load it (it works)
2. Go to the parent directory and try to load the same model with the same uri (throws that error).

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ .] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ .] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Edit : I'm 90% sure, this has something to do with pickling and unpickling of objects but what is the purpose of storing it in cloud when i need to keep a local copy too somewhere (correct me if i'm wrong tho)

**Comment by vibss2397** [comment]
It has to do with pickling storing relative paths to model files, so i would rather store the file as an artifact and have it downloaded to the same location or some workaround like that
